corehq/apps/locations/bulk_management.py

```diff
@@ -809,12 +809,21 @@ def update_children(parent_stub):
 
         for child_stub in child_stubs:
             child = child_stub.db_object
-            child.parent = parent_location
             if child_stub.do_delete:
                 # keep track of to be deleted items to delete them in top-to-bottom order
                 to_be_deleted.append(child)
             elif child_stub.needs_save:
                 child.location_type = types_by_code.get(child_stub.location_type)
+                if parent_location:
+                    # refetch parent_location object so that mptt related fields are updated consistently,
```

Comment:
Didn't use any of mptt methods (`delay_mptt_updates`, `rebuild` and `partial_rebuild`) because it gets challenging to sync couch changes as these depend on mptt data (which won't be update since we delay mptt). We would need to redo couch changes separately again.